roles/activemq/tasks/mask_password.yml

```diff
@@ -1,45 +1,66 @@
-        mask_pwd: ''
+    - name: "Set when requires to parse masked password salt"
       when:
         - not item.password is defined or not item.password is regex('^ENC\\([0-9]{4,}:[^:]+:[^:]+\\)')
+      block:
+        - name: "Parse passwd hash for existing user: {{ item.user }}"
+          ansible.builtin.set_fact:
+            existing_user: "{{ existing_users.content | b64decode
+                            | regex_search('\\b' + item.user + ' ?= ?ENC\\([0-9]+:([^:]+):([^)]+)\\)', '\\1', '\\2') | default([]) }}"
+            mask_pwd: ''
+          no_log: true
 
-    - name: Parse passwd for existing user salt
-      ansible.builtin.set_fact:
-        hash_password: "{{ item.password | middleware_automation.amq.pbkdf2_hmac(
-                           hashname=activemq_mask_password_hashname, iterations=activemq_mask_password_iterations, hexsalt=existing_user[0]) }}"
-      when: existing_user | length > 0
-      no_log: true
+        - name: "Parse passwd for existing user salt: {{ item.user }}"
+          ansible.builtin.set_fact:
+            hash_password: "{{ item.password | middleware_automation.amq.pbkdf2_hmac(
+                              hashname=activemq_mask_password_hashname,
+                              iterations=activemq_mask_password_iterations, hexsalt=existing_user[0]) }}"
+          when:
+            - existing_user is defined and existing_user | length > 0
+          no_log: true
 
-    - name: Get masked password for user
-      ansible.builtin.command: "{{ activemq.instance_home }}/bin/artemis mask --hash {{ '--password-codec' if activemq_password_codec != 'org.apache.activemq.artemis.utils.DefaultSensitiveStringCodec' else '' }} -- '{{ item.password }}'"
-      become: true
-      become_user: "{{ activemq_service_user }}"
-      register: mask_pwd
-      changed_when: false
-      no_log: true
-      when:
-        - not item.password is defined or not item.password is regex('^ENC\\([0-9]{4,}:[^:]+:[^:]+\\)')
-        - existing_user | length == 0 or hash_password != existing_user[1]
+        - name: Get masked password for user
+          ansible.builtin.command:
+            cmd: "{{ activemq.instance_home }}/bin/artemis mask --hash {{ '--password-codec' if activemq_password_codec != 'org.apache.activemq.artemis.utils.DefaultSensitiveStringCodec' else '' }} -- '{{ item.password }}'"
+          environment:
+            PATH: "{{ rpm_java_home | default(activemq_rpm_java_home, true) }}/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
+            JAVA_HOME: "{{ rpm_java_home }}"
+          become: true
+          become_user: "{{ activemq_service_user }}"
+          no_log: true
+          register: mask_pwd
+          changed_when: false
+          when:
+            - existing_user is not defined or existing_user | length == 0 or hash_password != existing_user[1]
```

Comment:
You'll never get there, the boolean logic is reversed with `or`; if existing_user is not defined the when exits immediately with true, otherwise hash_password will also be defined the third conditional will be evaluated without errors

Reply:
indeed, my bad. I had an error in my reproduction test

Reply:
no worries, actually, thank you for reviewing!